content/docs/rfcs/58/README.md

```diff
@@ -0,0 +1,189 @@
+---
+slug: 58
+title: 58/RLN-V2
+name: Rate Limit Nullifier V2
+status: draft
+editor: Rasul Ibragimov <curryrasul@gmail.com>
+contributors: 
+- Lev Soukhanov <0xdeadfae@gmail.com>
+---
+
+# Abstract
+
+The `58/RLN-V2` protocol is improvement of `32/RLN-V1`, being more general construct, that allows to set various limits for an epoch (it's 1 message per epoch in `32/RLN-V1`) while remaining almost as simple as it predecessor. Moreover, it allows to set different rate-limits for different RLN app users based on some public data, e.g. stake or reputation.
+
+# Motivation
+
+The main goal of `58/RLN-V2` is to generalize `32/RLN-V1` and expand its applications. There are two different subprotocols based on this protocol:
+* RLN-Same - RLN with the same rate-limit for different users;
+* RLN-Diff - RLN that allows to set different rate-limits for different users.
+
+It is important to note that by using large values ​​of N, users will be able to remain anonymous, because their `internal_nullifiers` will not be repeated until they exceed the limit.
```

Comment:
N hasn't been described yet here. Can we move this to the section where N is defined?